src/main/client/batch_write.c

```diff
@@ -173,6 +173,9 @@ static PyObject *AerospikeClient_BatchWriteInvoke(AerospikeClient *self,
         as_error_update(err, AEROSPIKE_ERR_PARAM,
                         "%s must be a list of BatchRecord",
                         FIELD_NAME_BATCH_RECORDS);
+        if (py_batch_records) {
+            Py_DECREF(py_batch_records);
+        }
```

Comment:
nit: Wouldn't a `Py_XDECREF` cover this without the if statement?

Reply:
After that if statement, it goes to label `CLEANUP4` which is after `Py_XDECREF(py_batch_records);`

```C
...
CLEANUP3:
    Py_XDECREF(py_batch_records);
CLEANUP4:
...
```

Looking back I could've used `CLEANUP3` but I can always refactor the code later

Reply:
The failing job is because of test noise-- the build passes and only one test case fails, usually only on macOS x86.